httpbakery/agent/discharge_test.go

```diff
@@ -0,0 +1,181 @@
+package agent_test
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"sync"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/macaroon.v1"
+
+	"gopkg.in/macaroon-bakery.v1/bakery"
+	"gopkg.in/macaroon-bakery.v1/bakery/checkers"
+	"gopkg.in/macaroon-bakery.v1/httpbakery"
+	"gopkg.in/macaroon-bakery.v1/httpbakery/agent"
+)
+
+type discharge struct {
+	cavId string
+	c     chan error
+}
+
+type Discharger struct {
```

Comment:
Thought for the future: it might be nice if the bakerytest discharger could support this kind of flexibility.